pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,408 @@
+        check_object_store_access
+        check_fluentd_connection
+        exit 0
+    fi
+    while [[ ${#} -gt  0 ]]; do
+      case ${1} in
+          --cluster-health )
+              check_opensearch_cluster_health
+          ;;
+          --snapshot-status )
+              check_opensearch_snapshots_status
+          ;;
+          --breakers )
+              check_opensearch_breakers
+          ;;
+          --indices )
+              check_opensearch_indices
+          ;;
+          --aliases )
+              check_opensearch_aliases
+          ;;
+          --mappings )
+              check_opensearch_mappings
+          ;;
+          --user-roles )
+              check_opensearch_user_roles
+          ;;
+          --ism )
+              check_opensearch_ism
+          ;;
+          --object-store-access )
+              check_object_store_access;
+          ;;
+          --fluentd )
+              check_fluentd_connection;
+          ;;
+          --help )
+            opensearch_check_help
+          ;;
+      esac
+      shift
+    done
+}
+
+check_opensearch_cluster_health() {
+  echo -ne "Checking if opensearch cluster is healthy ... "
+  cluster_health=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cluster/health")
+  status=$(echo "$cluster_health" | jq -r '.status')
+  if [[ $status != "green" ]]; then
+    echo -e "failure ❌";
+    echo "$cluster_health" | jq
+  else
+    echo -e "success ✔";
+  fi
+}
+
+check_opensearch_snapshots_status(){
+  echo -ne "Checking opensearch snapshots status ... "
+  no_error=true
+  debug_msg=""
+  repo_name=$(yq4 -e '.opensearch.snapshot.repository' "${config['config_file_sc']}")
+  repo_exists_status=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_snapshot/${repo_name}" | jq "select(.error)")
+  if [[ -z "$repo_exists_status" ]]; then
+    if kubectl get "cronjob" -n "opensearch-system" "opensearch-backup" &> /dev/null
+    then
+        last_schedule=$(kubectl get cronjob -n opensearch-system opensearch-backup -ojsonpath='{.status.lastScheduleTime}')
+        last_schedule_epoch_time=$(date -d "$last_schedule" +%s)
+        snapshots=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cat/snapshots/${repo_name}")
+        IFS=$'\n' readarray -t data < <(awk '{ print $1 " " $2 " " $3}' <<< "$snapshots")
+
+        for snapshot in "${data[@]}"; do
+          IFS=" " read -r -a snapshot_data <<< "$snapshot"
+          if [[ "${snapshot_data[1]}" == "FAILED" ]]; then
+            no_error=false
+            debug_msg+="[ERROR] The snapshot ${snapshot_data[0]} terminated in an error and no data was stored.\n"
+          fi
+        done
+
+        IFS=" " read -ra last_snapshot <<< "${data[-1]}"
+
+        if [[ ${last_schedule_epoch_time} -gt ${last_snapshot[2]}  ]]; then
+          no_error=false
+          debug_msg+="[ERROR] opensearch-backup cronjob has not been scheduled\n  "
+        fi
```

Comment:
What is actually checked here, that there is a snapshot done when it was asked to?

I think we should rework it to look at simply if the latest snapshot in opensearch was created within the configured window for snapshots and not based on the cronjob.
(We could simplify and say within 24 hours.)